app/services/companyReview.js

```diff
@@ -0,0 +1,40 @@
+const CrudService = require('./CrudService');
+const companyService = require('./company');
+const userService = require('./user');
+const companyReviewRepository = require('../repositories/companyReview');
+const error = require('./error');
+
+class CompanyReviewService extends CrudService {
+  async create(companyReview) {
+    if (companyReview.id || companyReview.companyId === undefined || companyReview.userId === undefined) {
+      error.throwValidationError('Invalid company review format.');
+    }
+
+    const companyReviewExists = await companyReviewRepository.getByPropsNonParanoid(companyReview);
+    if (companyReviewExists) {
+      error.throwValidationError('Company review already exists.');
+    }
+
+    await companyService.get(companyReview.companyId);
+    await userService.get(companyReview.userId);
+    const createdCompanyReview = await companyReviewRepository.create(companyReview);
+
+    return await this.get(createdCompanyReview.id);
+  }
+
+  async update(id, newData) {
+    if (newData.id !== undefined) {
```

Comment:
I wouldn't allow users to update their reviews. Once a review has been made, no one can update it or remove it. So please overwrite the delete method as well so that the reviews cannot be deleted.